### PR TITLE
python3Packages.imread: 0.7.0 -> 0.7.4, python3Packages.mahotas: 1.4.10 -> 1.4.11

### PR DESCRIPTION
**Pull request by wamserma**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/issues/124050

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 124362` at 5822e417 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/6a710fe585efca55a1558affaec95670">python38Packages.mahotas</a></li>
    <li><a href="https://gist.github.com/8e0ada3d08e890184b1d1d018c384c8f">python39Packages.mahotas</a></li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.imread</li>
    <li>python39Packages.imread</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-explicit-check-phase.md">python-explicit-check-phase</a>

Consider using `pytestCheckHook` in `checkInputs` rather than `checkPhase = "pytest";`.

Near pkgs/development/python-modules/mahotas/default.nix:29:3:
```
   |
29 |   checkPhase = ''
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by luispedro — APPROVED** [review]
LGTM.

File: pkgs/development/python-modules/mahotas/default.nix [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
+    sha256 = "029gvy1fb855pvxvy8zwj44k4s7qpqi0161bg5wldfiprrysn1kw";
   };
 
   # remove this as soon as https://github.com/luispedro/mahotas/issues/97 is fixed
   patches = [ ./disable-impure-tests.patch ];
 
-  propagatedBuildInputs = [ numpy imread pillow scipy ];
-  checkInputs = [ nose ];
+  propagatedBuildInputs = [ numpy imread pillow scipy freeimage ];
+  checkInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mahotas/default.nix:18]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

File: pkgs/development/python-modules/mahotas/default.nix [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
-  checkPhase= ''
-    python setup.py test
+  postPatch = ''
+    substituteInPlace mahotas/io/freeimage.py --replace "/opt/local/lib" "${freeimage}/lib"
+  '';
+
+  # tests must be run in the build directory
+  checkPhase = ''
+    cd build/lib*
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mahotas/default.nix:27]:
```suggestion
  # tests must be run in the build directory
  preCheck = ''
    cd build/lib*
```

File: pkgs/development/python-modules/mahotas/default.nix [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
   '';
 
   disabled = stdenv.isi686; # Failing tests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mahotas/default.nix:30]: Do they still fail?

File: pkgs/development/python-modules/mahotas/default.nix [hunk 4]
```diff
@@ -1,24 +1,30 @@
-{ buildPythonPackage, fetchFromGitHub, nose, pillow, scipy, numpy, imread, lib, stdenv }:
+{ buildPythonPackage, fetchFromGitHub, pillow, scipy, numpy, pytest, imread, freeimage, lib, stdenv }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mahotas/default.nix:1]:
```suggestion
{ buildPythonPackage, fetchFromGitHub, pillow, scipy, numpy, pytestCheckHook, imread, freeimage, lib, stdenv }:
```

File: pkgs/development/python-modules/mahotas/default.nix (hunk 3 above)
Comment (wamserma) [reply pkgs/development/python-modules/mahotas/default.nix:30]: I didn't check and rather would leave that to someone having the machinery in place.

File: pkgs/development/python-modules/mahotas/default.nix (hunk 2 above)
Comment (wamserma) [reply pkgs/development/python-modules/mahotas/default.nix]: updated PR

File: pkgs/development/python-modules/mahotas/default.nix (hunk 1 above)
Comment (wamserma) [reply pkgs/development/python-modules/mahotas/default.nix]: updated PR

File: pkgs/development/python-modules/mahotas/default.nix [hunk 5]
```diff
@@ -1,24 +1,29 @@
...
+    sha256 = "029gvy1fb855pvxvy8zwj44k4s7qpqi0161bg5wldfiprrysn1kw";
   };
 
   # remove this as soon as https://github.com/luispedro/mahotas/issues/97 is fixed
   patches = [ ./disable-impure-tests.patch ];
 
-  propagatedBuildInputs = [ numpy imread pillow scipy ];
-  checkInputs = [ nose ];
+  propagatedBuildInputs = [ numpy imread pillow scipy freeimage ];
+  checkInputs = [ pytestCheckHook ];
```
Comment (jonringer) [inline pkgs/development/python-modules/mahotas/default.nix:18]: could remove patch
```suggestion
  checkInputs = [ pytestCheckHook ];
  
  disabledTests = [
    "test_ellipse_axes"
    "test_normalize"
    "test_haralick3d"
  ];
```
and please add comment as to why they need to be disabled

Comment (wamserma) [reply pkgs/development/python-modules/mahotas/default.nix:15]: updated with your suggestions

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 124362` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python38Packages.imread</li>
<li>python38Packages.mahotas</li>
<li>python39Packages.imread</li>
<li>python39Packages.mahotas</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 124362` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python38Packages.imread</li>
<li>python38Packages.mahotas</li>
<li>python39Packages.imread</li>
<li>python39Packages.mahotas</li>
</ul>
</details>

**Pull request merged**